lib/loadObj.js

```diff
@@ -135,6 +152,94 @@ function loadObj(objPath, options) {
         primitive.material = getName(name);
     }
 
+    var intPoint = new Cartesian3();
+    var xAxis = Cesium.Cartesian3.UNIT_X.clone();
+    var yAxis = Cesium.Cartesian3.UNIT_Y.clone();
+    var zAxis = Cesium.Cartesian3.UNIT_Z.clone();
+    var origin = new Cartesian3();
+    var normal = new Cartesian3();
+    var ray = new Ray();
+    var plane = new Plane(Cesium.Cartesian3.UNIT_X, 0);
+    function projectTo2D(positions) {
+        var positions2D = new Array(positions.length);
```

Comment:
It would be best to use a scratch array for `positions2D` too. There should also be some bank of scratch `Cartesian2` that can be used in place of the allocations below. The same applies to the cartesians that are allocated for `positions3D`.